crates/matrix-sdk-crypto/src/machine/mod.rs

```diff
@@ -2378,6 +2378,33 @@ impl OlmMachine {
         DehydratedDevices { inner: self.to_owned() }
     }
 
+    /// Get the cached dehydrated device pickle key if any.
+    ///
+    /// None if the key was not previously cached (via
+    /// [`Self::save_dehydrated_device_pickle_key`]).
+    ///
+    /// Should be used to periodically rotate the dehydrated device to avoid
+    /// OTK exhaustion and accumulation of to_device messages.
+    pub async fn get_dehydrated_device_pickle_key(
+        &self,
+    ) -> StoreResult<Option<DehydratedDeviceKey>> {
+        self.inner.store.load_dehydrated_device_pickle_key().await
+    }
+
+    /// Store the dehydrated device pickle key in the crypto store.
+    ///
+    /// Use `None` to delete any previously saved pickle key.
```

Comment:
Doesn't that meant that we're going to accidentally delete the pickle key after every `receive_sync_changes()` call?

Reply:
Fixed, there is a specific API to delete the pickle key now